src/SAWScript/Crucible/MIR/Override.hs

```diff
@@ -301,23 +301,37 @@ matchArg opts sc cc cs prepost md actual expectedTy expected@(MS.SetupTerm expec
        realTerm <- valueToSC sym md failMsg tval actual
        matchTerm sc cc md prepost realTerm (ttTerm expectedTT)
 
-matchArg opts sc cc cs _prepost md actual@(MIRVal (RefShape _refTy pointeeTy mutbl tpr) ref) expectedTy setupval =
-  case setupval of
-    MS.SetupVar var ->
+matchArg opts sc cc cs prepost md actual expectedTy expected =
+  mccWithBackend cc $ \bak -> do
+  let sym = backendGetSym bak
+  case (actual, expectedTy, expected) of
+    (MIRVal (RefShape _refTy pointeeTy mutbl tpr) ref, _, MS.SetupVar var) ->
       do assignVar cc md var (Some (MirPointer tpr mutbl pointeeTy ref))
 
-    MS.SetupNull empty                -> absurd empty
-    MS.SetupGlobal empty _            -> absurd empty
-    MS.SetupCast empty _              -> absurd empty
-    MS.SetupUnion empty _ _           -> absurd empty
-    MS.SetupGlobalInitializer empty _ -> absurd empty
-
-    _ -> failure (cs ^. MS.csLoc) =<<
-           mkStructuralMismatch opts cc sc cs actual setupval expectedTy
-
-matchArg opts sc cc cs _prepost md actual expectedTy expected =
-  failure (MS.conditionLoc md) =<<
-    mkStructuralMismatch opts cc sc cs actual expected expectedTy
+    -- match arrays point-wise
+    (MIRVal (ArrayShape _ _ elemShp) xs, Mir.TyArray y _len, MS.SetupArray _elemTy zs)
+      | Mir.MirVector_Vector xs' <- xs
+      , V.length xs' == length zs ->
+        sequence_
+          [ matchArg opts sc cc cs prepost md (MIRVal elemShp x) y z
+          | (x, z) <- zip (V.toList xs') zs ]
+
+      | Mir.MirVector_PartialVector xs' <- xs
+      , V.length xs' == length zs ->
+        do xs'' <- liftIO $
+             traverse (readMaybeType sym "vector element" (shapeType elemShp)) xs'
+           sequence_
+             [ matchArg opts sc cc cs prepost md (MIRVal elemShp x) y z
+             | (x, z) <- zip (V.toList xs'') zs ]
```

Comment:
Here is how SAW matches array arguments in specifications. Note that `MirVector_Array` (for symbolic, SMT-based arrays) aren't supported.